### PR TITLE
IGNITE-13992 Migrates Spring Transactions integration to ignite-extensions.

### PR DESCRIPTION
**Pull request by ololo3000**

### CODE REVIEW

File: modules/spring-data-commons/src/main/java/org/apache/ignite/internal/transactions/proxy/ClientTransactionProxy.java [hunk 1]
```diff
@@ -0,0 +1,60 @@
...
+package org.apache.ignite.internal.transactions.proxy;
+
+import org.apache.ignite.client.ClientTransaction;
+import org.apache.ignite.internal.util.typedef.internal.S;
+
+/**
+ * Represents {@link TransactionProxy} implementation that uses {@link ClientTransaction} to perform transaction
+ * operations.
+ */
+public class ClientTransactionProxy implements TransactionProxy {
```
Comment (alex-plekhanov) [inline modules/spring-data-commons/src/main/java/org/apache/ignite/internal/transactions/proxy/ClientTransactionProxy.java:27]: As far as I understand transaction proxies don't use anyhow cache data proxies and cache data proxies don't use anyhow transactions proxies. Transactions proxies required only by one module spring-tx, this module can be used without spring data, so I think we should move back these classes to spring-tx module.
Moreover, there is no dependency from spring-tx to spring-data-commons module.

Comment (ololo3000) [reply modules/spring-data-commons/src/main/java/org/apache/ignite/internal/transactions/proxy/ClientTransactionProxy.java:27]: My motivation to put the tx proxy classes in spring-data-commons is to make it possible to reuse all the proxy classes.

In the future, I would like to rename spring-data-commons to spring-commons.

I think it is useful to keep all of the Spring utility classes in one module - for example, the IgniteCacheProxy, which is now stored in spring-data-commons, will be reused in  the Spring Cache thin client integration. But earlier I claimed that IgniteCacheProxy is only needed for Spring Data. The same can be with the tx proxy.

Does this make sense?

Comment (alex-plekhanov) [reply modules/spring-data-commons/src/main/java/org/apache/ignite/internal/transactions/proxy/ClientTransactionProxy.java:27]: Spring data proxies are used by 3 modules, so it makes sense to move them into the common module to reduce duplication. Spring-tx proxies only used by one. If in the future there will be new modules that will require spring-tx proxies, we can move these classes to the common module, but now I don't know modules except spring-tx that require spring-tx proxies. Did I miss something?

Comment (ololo3000) [reply modules/spring-data-commons/src/main/java/org/apache/ignite/internal/transactions/proxy/ClientTransactionProxy.java]: I moved proxy classes back to tx-ext module.

**Review by alex-plekhanov — APPROVED** [review]

**Pull request closed without merging**